ansible_base/jwt_consumer/common/auth.py

```diff
@@ -34,32 +39,46 @@
 class JWTCommonAuth:
     def __init__(self, user_fields=default_mapped_user_fields) -> None:
         self.mapped_user_fields = user_fields
+        self.user = None
+        self.token = None
+        self.team_content_type = ContentType.objects.get_for_model(apps.get_model(settings.ANSIBLE_BASE_TEAM_MODEL))
+        self.org_content_type = ContentType.objects.get_for_model(apps.get_model(settings.ANSIBLE_BASE_ORGANIZATION_MODEL))
 
     def get_cache_timeout(self):
         # If unspecified the cache will expire in 7 days
         cache_timeout = get_setting('ANSIBLE_BASE_JWT_CACHE_TIMEOUT_SECONDS', 604800)
         return cache_timeout
 
     def parse_jwt_token(self, request):
+        """
+        parses the given request setting self.user and self.token
+        """
+
+        self.user = None
+        self.token = None
+
         logger.debug("Starting JWT Authentication")
-        token = request.headers.get("X-DAB-JW-TOKEN", None)
-        if not token:
+        if request is None:
+            return
+
+        unencrypted_token = request.headers.get("X-DAB-JW-TOKEN", None)
```

Comment:
Is this more accurately undecoded_token?  Isn't this code decrypting it?  Or maybe it was already decrypted in some way..

Reply:
No, good catch, that needs a better name. Its probably encoded and encrypted when its in the header.